src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,64 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+
+import { RazorLogger } from './RazorLogger';
+
+export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+    private logger: RazorLogger;
+    private vscodeType: typeof vscode;
+
+    constructor(logger: RazorLogger, vscodeType: typeof vscode) {
+        this.logger = logger;
+        this.vscodeType = vscodeType;
+    }
+
+    public resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, configuration: vscode.DebugConfiguration): vscode.ProviderResult<vscode.DebugConfiguration> {
+        const app = {
+            name: '.NET Core Launch (Blazor Standalone)',
+            type: 'coreclr',
+            request: 'launch',
+            program: 'dotnet',
+            args: ['run'],
+            cwd: '${workspaceFolder}',
+            env: {
+                ASPNETCORE_ENVIRONMENT: 'Development',
+                ...configuration.env,
+            },
+        };
+        const browser = {
+            name: '.NET Core Debug Blazor Web Assembly in Browser',
+            type: configuration.browser === 'edge' ? 'edge' : 'pwa-chrome',
+            request: 'launch',
+            timeout: 30000,
+            url: configuration.url || 'https://localhost:5001',
+            webRoot: '${workspaceFolder}',
+            inspectUri: '{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}',
+        };
+
+        this.vscodeType.debug.startDebugging(folder, app).then(
+            appStartFulfilled => {
+                if (appStartFulfilled) {
+                    this.vscodeType.debug.startDebugging(folder, browser).then(
+                        debugStartFulfilled => {
+                            if (debugStartFulfilled) {
+                                this.logger.logVerbose('Launching JavaScript debugger...');
+                            }
+                        },
+                        error => {
+                            this.logger.logError('Error when launching Chrome debugger: ', error);
+                        },
+                    );
+                }
+            },
+            error => {
+                this.logger.logError('Error when launching application: ', error);

```

Comment:
Since we're adding debug adapter support to this extension it might make sense to expand our `RazorLogger` impl to support scopes. Aka any debugging log should probably have the `[DEBUGGER]` prefix to ensure those logs can be differentiated from any Razor editor logs.

Reply:
I've added this prefix manually for the time being.

I created https://github.com/dotnet/aspnetcore/issues/21715 to track this.